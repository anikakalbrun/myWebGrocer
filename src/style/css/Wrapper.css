.hero__vertical-container {
    position: relative;
}

.hero__title {
    position: fixed;
    top: 50%;
    left: 50%;
    -webkit-transform: translate(-50%,-50%);
    transform: translate(-50%,-50%);
    background-color: rgba(59,200,76,.9);
    color: #fff;
    font-weight: 300;
    padding: 4rem;
    max-width: 550px;
}

.hero__title.large {
    max-width: 2000px;
    min-width: 1000px;
}

.hero__title.inverse {
    background-color: rgba(255,255,255,.95);
    color: #43c653;
}

.hero__title.inverse.default {
    background-color: rgba(255,255,255,.95);
    color: #222222;
}

@media (max-width: 991px) {
    .hero__title {
        min-width: 80%;
        top: 400px;
        left: 50%;
        padding: 2.625rem;
    }
    .hero__title.inverse {
        min-width: 60%;
    }
}

@media (max-width: 400px) {
    .hero__title.inverse {
        min-width: 80%;
    }
}