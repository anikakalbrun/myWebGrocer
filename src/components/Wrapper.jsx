import React from 'react';
import {string, node} from 'prop-types';

import '../style/css/Wrapper.css';

Wrapper.propTypes = {
    children: node,
    innerContainerClass: string
};

Wrapper.defaultProps = {
    innerContainerClass: ''
};

export default function Wrapper({children, innerContainerClass}) {
    return (
        <div className="hero__vertical-container">
            <div className={`hero__title ${innerContainerClass}`}>
                {children}
            </div>
        </div>
    );
}