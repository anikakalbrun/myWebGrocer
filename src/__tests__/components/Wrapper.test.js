import React from 'react';
import Wrapper from '../../components/Wrapper';

let wrapper;
const props = {
    children: <p/>,
    innerContainerClass: 'green'
};

describe('Wrapper', () => {

    it('renders correctly', () => {
        wrapper = renderer.create(<Wrapper {...props}/>).toJSON();
        expect(wrapper).toMatchSnapshot();
    });

    it('defaultProps includes innerContainerClass that is equal to ""', () => {
        wrapper = renderer.create(<Wrapper/>).toJSON();
        expect(wrapper).toMatchSnapshot();
    });
});
