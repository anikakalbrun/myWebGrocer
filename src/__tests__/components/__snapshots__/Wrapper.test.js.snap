// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper defaultProps includes innerContainerClass that is equal to "" 1`] = `
<div
  className="hero__vertical-container"
>
  <div
    className="hero__title "
  />
</div>
`;

exports[`Wrapper renders correctly 1`] = `
<div
  className="hero__vertical-container"
>
  <div
    className="hero__title green"
  >
    <p />
  </div>
</div>
`;
